crucible-llvm/src/Lang/Crucible/LLVM/Translation.hs

```diff
@@ -152,298 +159,36 @@ import           Lang.Crucible.Types
 
 type ModuleCFGMap arch = Map L.Symbol (C.AnyCFG (LLVM arch))
 
+-- | A @GlobalInitializerMap@ records the initialized values of globals in an @L.Module@.
+--
+-- The @Left@ constructor is used to signal errors in translation,
+-- which can happen when:
+-- * The global isn't actually a compile-time constant
+-- * The declaration is ill-typed
+-- * The global isn't linked ("extern global")
+--
+-- These failures are as granular as possible (attached to the values)
+-- so that simulation still succeeds if the module has a bad global that the
+-- verified function never touches.
+--
+-- To actually initialize globals, saw-script translates them into
+-- instances of @MemModel.LLVMVal@.
+type GlobalInitializerMap = Map L.Symbol (Either String LLVMConst)
```

Comment:
This map should probably contain more information than it does (optimally, anything someone down the road would want to know about a given global). The least "lossy" way of doing this is probably
```haskell
type GlobalInitializerMap = Map L.Global (Either String LLVMConst)
```